game-engine/src/commonMain/kotlin/za/co/entelect/challenge/game/engine/renderer/printables/PrintablePlayer.kt

```diff
@@ -25,15 +29,38 @@ class PrintablePlayer(player: WormsPlayer) {
          * Build a PrintablePlayer from @player that is modified to fit the perspective of @perspectivePlayer
          * @perspectivePlayer is not allowed to see some details from other players
          */
-        fun buildForPerspectivePlayer(player: WormsPlayer, perspectivePlayer: WormsPlayer): PrintablePlayer {
+        fun buildForPerspectivePlayer(player: WormsPlayer, perspectivePlayer: WormsPlayer, wormsMap: WormsMap): PrintablePlayer {
             val playerForPerspectivePlayer = PrintablePlayer(player)
             playerForPerspectivePlayer.worms = player.worms
                     .map { PrintableWorm.buildForDetailsPerspectivePlayer(it, perspectivePlayer) }
             if (isPerspectivePlayer(player, perspectivePlayer)) {
                 playerForPerspectivePlayer.health = player.health
             }
+            playerForPerspectivePlayer.previousCommand = getLastCommand(wormsMap, player)
             return playerForPerspectivePlayer
         }
+
+        private fun getLastCommand(wormsMap: WormsMap, player: WormsPlayer): String {
+            val feedback = wormsMap
+                .getFeedback(wormsMap.currentRound - 1)
+                .filter { it.playerId == player.id }
+            val feedbackCount = feedback.size
+            return when {
+                (feedbackCount == 1) -> feedback.get(0).command
+                (feedbackCount == 2) -> extractSelectCommand(feedback)
+                else                 -> CommandStrings.NOTHING
+            }
+        }
+
+        private fun extractSelectCommand(feedback: List<CommandFeedback>): String {
+            val selectCommand = feedback
+                .filter { it.command.startsWith("select") }
+                .get(0)
```

Comment:
Try this instead
```kotlin
val selectCommand = feedback.find { it.command.startsWith("select") }
```

Reply:
or `.first`, they handle nulls differently

Reply:
Thanks.  I searched the docs to try and find the `find` function (lol) (because it's quite common in programming languages with these sorts of iterators) and couldn't find it XD